components/eam/src/chemistry/utils/yaml_input_file_io.F90

```diff
@@ -974,10 +979,20 @@ subroutine write_1d_var_real(unit_input, unit_output, fld_name,dim,field)
 
     write(unit_input,'(3A)',advance="no")'    ',trim(adjustl(fld_name)),': ['
 
-    write(unit_input,10,advance="no")field(lbound(field,1))
+    if(isnan(field(lbound(field,1)))) then
```

Comment:
Would you please try if the `where` clause works here? The code should look something like this:
`field_mod=field`
`where (isnan(field_mod)) field_mod = realnanval`

This code should be added just after the `format` statements. `field_mod` is a local variable to store `field` info as we can't change intent-in `field`

Reply:
I use a function 'remove_nan_real' now to handle this.  The input to the function is a particular element within an array, not the whole array, so 'where' is not needed.  This also bypasses the need for a storage variable 'field_mod'.

Reply:
Great solution! Thanks, Brian!